lib/maths/CBoostedTreeFactory.cc

```diff
@@ -276,25 +276,32 @@ CBoostedTreeFactory::constructFromParameters(std::size_t numberThreads,
 CBoostedTreeFactory::TBoostedTreeUPtr
 CBoostedTreeFactory::constructFromString(std::stringstream& jsonStringStream,
                                          core::CDataFrame& frame) {
-    TBoostedTreeUPtr treePtr{
-        new CBoostedTree{frame, TBoostedTreeImplUPtr{new CBoostedTreeImpl{}}}};
-    core::CJsonStateRestoreTraverser traverser(jsonStringStream);
-    treePtr->acceptRestoreTraverser(traverser);
-    return treePtr;
+    try {
+        TBoostedTreeUPtr treePtr{
+            new CBoostedTree{frame, TBoostedTreeImplUPtr{new CBoostedTreeImpl{}}}};
+        core::CJsonStateRestoreTraverser traverser(jsonStringStream);
+        if (treePtr->acceptRestoreTraverser(traverser) == false) {
```

Comment:
Should probably be:

```
    if (treePtr->acceptRestoreTraverser(traverser) == false ||
        traverser.haveBadState()) {
```

I think this might be needed to pick up every possible error.  It would match: https://github.com/elastic/ml-cpp/blob/55a00f7004a55eca9bf6188190a8125679b08ef6/lib/api/CAnomalyJob.cc#L961-L962

Reply:
Done in [b3f6533](https://github.com/elastic/ml-cpp/pull/577/commits/b3f65336b92343c73d410e44998482efd0840906).